htlcswitch/hop/payload.go

```diff
@@ -222,20 +222,51 @@ func NewPayloadFromReader(r io.Reader, finalHop bool,
 	// Filter out the custom records.
 	customRecords := NewCustomRecords(parsedTypes)
 
-	return &Payload{
-		FwdInfo: ForwardingInfo{
-			NextHop:         lnwire.NewShortChanIDFromInt(cid),
-			AmountToForward: lnwire.MilliSatoshi(amt),
-			OutgoingCTLV:    cltv,
-		},
+	// If we have a blinding point, pick it from our two options (we've
+	// already validated that we don't have _both_ set).
+	var activeBlindingPoint *btcec.PublicKey
```

Comment:
What about if _neither_ is set here?

Reply:
We've previously validated that we have _only one_ if we have `encrypted_data` present (per bolt 04), but the refactor you suggested makes this cleaner 👌